test/karma-test-shim.js

```diff
@@ -87,9 +87,38 @@ function configureTestBed() {
 
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
 
-    testing.TestBed.initTestEnvironment(
+    var testBed = testing.TestBed.initTestEnvironment(
       testingBrowser.BrowserDynamicTestingModule,
       testingBrowser.platformBrowserDynamicTesting()
     );
+
+    applyAngularTestingPatches(testBed);
+  });
+}
+
+/** Applies patches and workarounds to Angular's testing package. */
+function applyAngularTestingPatches(testBed) {
```

Comment:
How about `patchTestModuleToDestroyFixturesAfterEveryTest`?

```js
/**
 * Monkey-patches TestBed.resetTestingModule such that any errors that occur during component
 * destruction are thrown instead of silently logged. Also runs TestBed.resetTestingModule after
 * each unit test.
 *
 * Without this patch, the combination of two behaviors is problematic for Angular Material:
 * - TestBed.resetTestingModule catches errors thrown on fixture destruction and logs them without
 *     the errors ever being thrown. This means that any component errors that occur in ngOnDestroy
 *     can encounter errors silently and still pass unit tests.
 * - TestBed.resetTestingModule is only called *before* a test is run, meaning that even *if* the
 *    aforementioned errors were thrown, they would be reported for the wrong test (the test that's
 *    about to start, not the test that just finished).
 */
function patchTestBedToDestroyFixturesAfterEveryTest() {
}
```